src/Searchbar.vue

```diff
@@ -15,41 +19,66 @@ export default {
       if (!this.data) {
         return undefined;
       }
-      const matches = [];
+      const pages = [];
       const regexes = this.value.split(' ')
         .filter(searchKeyword => searchKeyword !== '')
         .map(searchKeyword => searchKeyword.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'))
-        .map(searchKeyword => new RegExp(searchKeyword, 'i'));
+        .map(searchKeyword => new RegExp(searchKeyword, 'ig'));
       this.data.forEach((entry) => {
-        const { headings, src, title } = entry;
+        const {
+          headings,
+          src,
+          title,
+          headingKeywords,
+        } = entry;
         const keywords = entry.keywords || '';
-        let searchTarget = [title].concat(keywords).concat(Object.values(headings)).join(' ');
-        let totalMatches = getTotalMatches(searchTarget, regexes);
-        if (totalMatches > 0) {
-          searchTarget = [title].concat(keywords).join(' ');
-          const isMatchingPage = getTotalMatches(searchTarget, regexes) === totalMatches;
-          if (isMatchingPage) {
-            matches.push(Object.assign(entry, { totalMatches }));
-          }
+        const fallbackTitle = title || src;
+
+        let searchTarget = [
+          fallbackTitle,
+          keywords,
+          ...Object.values(headings),
+          ...Object.values(headingKeywords),
+        ].join(' ');
```

Comment:
It feels like it would be nice to move the `join(' ')` part to `getTotalMatches`, so that we can just pass a list of search targets to the function instead of having to join it manually every time.